01_Data/src/layers/sequelize/repository/LocalAuthList.ts

```diff
@@ -3,196 +3,209 @@
+
+  constructor(
+    config: SystemConfig,
+    logger?: Logger<ILogObj>,
+    sequelizeInstance?: Sequelize,
+    authorization?: IAuthorizationRepository,
+    localListAuthorization?: CrudRepository<LocalListAuthorization>,
+    sendLocalList?: CrudRepository<SendLocalList>,
+  ) {
+    super(config, Authorization.MODEL_NAME, logger, sequelizeInstance);
+    this.authorization = authorization ? authorization : new SequelizeAuthorizationRepository(config, logger);
+    this.localListAuthorization = localListAuthorization ? localListAuthorization : new SequelizeRepository<LocalListAuthorization>(config, LocalListAuthorization.MODEL_NAME, logger, sequelizeInstance);
+    this.sendLocalList = sendLocalList ? sendLocalList : new SequelizeRepository<SendLocalList>(config, SendLocalList.MODEL_NAME, logger, sequelizeInstance);
+  }
+
+  async createSendLocalListFromRequestData(stationId: string, correlationId: string, updateType: UpdateEnumType, versionNumber: number, localAuthorizationList?: AuthorizationData[]): Promise<SendLocalList> {
+    const sendLocalList = await SendLocalList.create({
+      stationId,
+      correlationId,
+      versionNumber,
+      updateType,
+    });
+    for (const authData of localAuthorizationList ?? []) {
+      const auth = await Authorization.findOne({
+        include: [
+          {
+            model: IdToken,
+            where: {
+              idToken: authData.idToken.idToken,
+              type: authData.idToken.type,
+            },
+          },
+          {
+            model: IdTokenInfo,
+            include: [IdToken],
+          },
+        ],
+      });
+      if (!auth) {
+        throw new Error(`Authorization not found for ${JSON.stringify(authData)}, invalid SendLocalListRequest (create necessary Authorizations first)`);
+      }
+      if (!deepDirectionalEqual(authData.idToken, auth.idToken)) {
+        throw new Error(`Authorization idToken in SendLocalListRequest ${JSON.stringify(authData.idToken)} does not match idToken in database ${JSON.stringify(auth.idToken)} (update the idToken first)`);
+      }
+      if (authData.idTokenInfo?.groupIdToken && (!auth.idTokenInfo?.groupIdToken || !deepDirectionalEqual(authData.idTokenInfo.groupIdToken, auth.idTokenInfo.groupIdToken))) {
+        throw new Error(
+          `Authorization group idToken in SendLocalListRequest ${JSON.stringify(authData.idTokenInfo.groupIdToken)} does not match group idToken in database ${JSON.stringify(auth.idTokenInfo?.groupIdToken)} (update the group idToken first)`,
+        );
+      }
+      // While new IdTokens will NOT be created or newly associated via message api, idTokenInfo can be allowed to be unique for the local auth list
+      const localListAuthIdTokenInfo = await IdTokenInfo.create({
+        ...authData.idTokenInfo,
+        groupIdTokenId: auth.idTokenInfo?.groupIdTokenId,
+      });
+
+      const { id, _idTokenInfoId, _idTokenInfo, ...authorizationFields } = auth;
```

Comment:
@elliot-sabitov why the `_` prefix? This breaks the build. I have a commit I can push to fix this. But wanted to ask if this was intentional.

Reply:
not intentional, I will adjust, this is an error, looks like those two fields arent being used and lint fails so I incorrectly prefixed with underscore by mistake, great catch thank you so much! 😮‍💨 Not sure how I missed that